<?xml version="1.0"?>
<robot name="PF400_Robot" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <!-- Materials -->

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>

  <material name="aluminum">
    <color rgba="0.6 0.6 0.6 1"/>
  </material>

  <!-- PF400 -->
  <xacro:include filename="$(find pf400_description)/urdf/test.urdf.xacro" />
  <xacro:pf400 prefix="${prefix}pf400/"/>
  

  <!-- Joints -->

  <joint name="J1" type="prismatic">
    <parent link="${prefix}pf400/z_column"/>
    <child link="${prefix}pf400/z_carriage"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0.03005 -0.0198 0.17535" rpy="0 0 0"/>
    <limit effort="1000.0" lower="0.002" upper="1.160" velocity="0.5"/>  </joint>

  <joint name="J2" type="revolute">
    <parent link="${prefix}pf400/z_carriage"/>
    <child link="${prefix}pf400/inner_link"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0.173001 0.020 0.0235" rpy="0 0 0"/>
    <limit effort="1000.0" lower="-1.62316" upper="1.62316" velocity="0.5"/>
  </joint>

  <joint name="J3" type="revolute">
    <parent link="${prefix}pf400/inner_link"/>
    <child link="${prefix}pf400/outer_link"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0.302 0 -0.072285" rpy="0 0 0"/>
    <limit effort="1000.0" lower="0.175" upper="6.1" velocity="0.5"/>
  </joint>

  <joint name="J4" type="revolute">
    <parent link="${prefix}pf400/outer_link"/>
    <child link="${prefix}pf400/hand"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0.289 0 -0.032942" rpy="0 0 0"/>
    <limit effort="1000.0" lower="-16.7552" upper="16.7552" velocity="0.5"/>
  </joint>

  <joint name="J5" type="prismatic">
    <parent link="${prefix}pf400/hand"/>
    <child link="${prefix}pf400/left_finger"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0.078573 0 -0.04269" rpy="0 0 0"/>
    <limit effort="1000.0" lower="0.067454" upper="0.070" velocity="0.5"/>
  </joint>

  <joint name="J5_mirror" type="prismatic">
    <parent link="${prefix}pf400/hand"/>
    <child link="${prefix}pf400/right_finger"/>
    <axis xyz="0 -1 0"/>
    <origin xyz="0.078573 0 -0.04269" rpy="0 0 0"/>
    <limit effort="1000.0" lower="0.067454" upper="0.070" velocity="0.5"/>
    <mimic joint="J5" multiplier="1"/>
  </joint>

  <joint name="J6" type="prismatic">
    <parent link="${prefix}pf400/rail_base"/>
    <child link="${prefix}pf400/rail"/>
    <axis xyz="1 0 0"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <limit effort="1000.0" lower="-1" upper="1" velocity="0.5"/>
  </joint>

  <joint name="rail_link" type="fixed">
    <parent link="${prefix}pf400/rail"/>
    <child link="${prefix}pf400/z_column"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0.03005 -0.0198 0.05" rpy="0 0 0"/>
    <limit effort="1000.0" lower="0" upper="0" velocity="0.5"/>
  </joint>

</robot>
